shopfloor/actions/search.py

```diff
@@ -34,12 +34,12 @@ def picking_from_scan(self, barcode):
             return model.browse()
         return model.search([("name", "=", barcode)], limit=1)
 
-    def product_from_scan(self, barcode):
+    def product_from_scan(self, barcode, use_packaging=True):
         model = self.env["product.product"]
         if not barcode:
             return model.browse()
         product = model.search([("barcode", "=", barcode)], limit=1)
-        if not product:
+        if not product and use_packaging:
             packaging = self.env["product.packaging"].search(
                 [("product_id", "!=", False), ("barcode", "=", barcode)], limit=1
             )
```

Comment:
Note for later: need to address comment done there: https://github.com/OCA/wms/pull/398#discussion_r865988306=